### PR TITLE
Bugfix/issue 470 - Suboptimal Multiplexing Behavior

### PR DESCRIPTION
**Pull request by askirk**
In reference to issue #470 ...

Fixing the case where the apps using a router service fall back to legacy BT when that router service is force closed. Now we check if there are additional Sdl Broadcast Receivers (and Router Services) that can be woken before falling back to legacy BT.

### CODE REVIEW

**Comment by askirk** [comment]
I'm not sure if the `isAdditionalSdlBroadcastReceiver()` method should check that a found BroadcastReceiver is verified using a method like `isTrustedPackage()`. Currently it does not.

**Comment by brettywhite** [comment]
I'd assume it should 🔎

File: sdl_android/src/main/java/com/smartdevicelink/SdlConnection/SdlConnection.java [hunk 1]
```diff
@@ -98,12 +98,17 @@ private void constructor(BaseTransportConfig transportConfig,RouterServiceValida
...
 						cachedMultiConfig = (MultiplexTransportConfig) transportConfig;
 						cachedMultiConfig.setService(null);
 					}
-					enableLegacyMode(true,TransportType.BLUETOOTH); //We will use legacy bluetooth connection for this attempt
+
+					if(!rsvp.isAdditionalSdlBroadcastReceiver()) { // If there are no other Router Services to wake up
+						Log.w(TAG, "SDL Router service isn't trusted. Enabling legacy bluetooth connection.");
+						enableLegacyMode(true, TransportType.BLUETOOTH); //We will use legacy bluetooth connection for this attempt
+					}else{
+						Log.w(TAG, "SDL Router service isn't trusted. Attempting to wake up other router services.");
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/SdlConnection/SdlConnection.java]: This  comment doesn't make sense here.

File: sdl_android/src/main/java/com/smartdevicelink/transport/RouterServiceValidator.java [hunk 2]
```diff
@@ -513,7 +513,7 @@ public boolean isAdditionalSdlBroadcastReceiver(){
 			String packageName = resolveInfo.activityInfo.packageName;
 			if(packageName != null){
 				if(lastClosedService != null){
-					if(isTrustedPackage(packageName, manager) && !lastClosedService.equals(packageName)){
+					if(isTrustedPackage(packageName, manager) && !lastClosedService.toString().equals(packageName)){
 						return true;
```
Comment (Tnnnguyen) [inline sdl_android/src/main/java/com/smartdevicelink/transport/RouterServiceValidator.java]: This condition check will not work the way it's intent to:
`!lastClosedService.toString().equals(packageName)`

The reason is that `lastClosedService` is a ComponentName and its toString method returns: 
`"ComponentInfo{" + mPackage + "/" + mClass + "}"`
which will never equal to `packageName`.

I think it should be: `!lastClosedService.getPackageName().equals(packageName)` if the intention is to exclude the current (last closed) service.

**Comment by timeyinxingren** [comment]
this problem has been fixed? @BrettyWhite

**Pull request closed without merging**